pkg/scheduler/scheduler_test.go

```diff
@@ -63,3 +69,211 @@ func TestCreateScheduler(t *testing.T) {
+		expected              *workv1alpha2.ResourceBinding
+	}{
+		{
+			name: "add success condition",
+			binding: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-1", Namespace: "default", Generation: 1},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{},
+			},
+			newScheduledCondition: successCondition,
+			expected: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-1", Namespace: "default", Generation: 1},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{successCondition}, SchedulerObservedGeneration: 1},
+			},
+		},
+		{
+			name: "add failure condition",
+			binding: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-2", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{},
+			},
+			newScheduledCondition: failureCondition,
+			expected: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-2", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{failureCondition}},
+			},
+		},
+		{
+			name: "replace to success condition",
+			binding: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-3", Namespace: "default", Generation: 1},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{failureCondition}, SchedulerObservedGeneration: 2},
+			},
+			newScheduledCondition: successCondition,
+			expected: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-3", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{successCondition}, SchedulerObservedGeneration: 1},
+			},
+		},
+		{
+			name: "replace failure condition",
+			binding: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-4", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{successCondition}},
+			},
+			newScheduledCondition: failureCondition,
+			expected: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-4", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{failureCondition}},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := karmadaClient.WorkV1alpha2().ResourceBindings(test.binding.Namespace).Create(context.TODO(), test.binding, metav1.CreateOptions{})
+			if err != nil {
+				t.Error(err)
```

Comment:
We should use `Fatal` or `Fatalf` here since if there is an error, we can skip following checks.